packages/ERTP/src/amountMath.chainmail

```diff
@@ -0,0 +1,117 @@
+/**
+ * Amounts are descriptions of digital assets, answering the questions
+ * "how much" and "of what kind". Amounts are branded extents. 
+ * AmountMath executes the logic of how amounts are changed when digital 
+ * assets are merged, separated, or otherwise manipulated. For
+ * example, a deposit of 2 bucks into a purse that already has 3 bucks
+ * gives a new balance of 5 bucks. An empty purse has 0 bucks. AmountMath 
+ * relies heavily on polymorphic MathHelpers, which manipulate the unbranded
+ * portion.
+ */
+struct Amount (Brand, Extent) {
+  brand :Brand;
+  extent :Extent;
+}
+
+/**
+ * Extents describe the extent of something that can be owned or shared.
+ * Fungible extents are normally represented by natural numbers. Other 
+ * extents may be represented as strings naming a particular right, or 
+ * an arbitrary object that sensibly represents the rights at issue. 
+ *
+ * Extent must be Comparable. (This IDL doesn't yet provide a way to specify
+ * subtype relationships for structs.)
+ */
+struct Extent {
+}
+
+/**
+ * Logic for manipulating amounts.
+ *
+ * Amounts are the canonical description of tradable goods. They are manipulated
+ * by issuers and mints, and represent the goods and currency carried by purses and
+ * payments. They can be used to represent things like currency, stock, and the
+ * abstract right to participate in a particular exchange.
+ *
+ */
+interface AmountMath (Amount (Brand, Extent)) {
+
+  /** Return the brand. */
+  getBrand() -> (Brand);
+
+  /** Get the name of the mathHelpers used. */
+  getMathHelpersName -> (String);
+
+  /**  Make an amount from an extent by adding the brand. */
+  make(allegedExtent :Extent) -> (Amount);
+
+  /**
+   * Make sure this amount (or extent) is valid and return it if so.
+   */
+  coerce(allegedAmountOrExtent :Amount) -> (Amount);
+
+  /** Remove the brand and return the extent. */
+  extent(amount :Amount) -> (Extent);
+
+  /** Return the amount representing an empty amount. This is the
+   * identity element in MathHelpers.
+   */
+  getEmpty() -> (Amount);
```

Comment:
For pass-by-copy objects, we should avoid being very specific about their object identity. The empty amount is the *algebraic* identity of the `add` operation. It has nothing to do with object identity. Since both terms are well entrenched, we cannot avoid the collision here.

We normally implement it by sharing an empty object, but we should not specify that. Any client code that depends on the object identity of pass-by-copy objects is likely incorrect.

Reply:
`amountMath.isEqual(amountMath.getEmpty(), amountMath.getEmpty())` will always be true, but no other equality is guaranteed.